src/sql/src/names.rs

```diff
@@ -53,6 +55,18 @@ impl From<Option<String>> for DatabaseSpecifier {
     }
 }
 
+impl From<&FullName> for ObjectName {
+    fn from(full_name: &FullName) -> ObjectName {
+        let mut name_parts = Vec::new();
+        if let DatabaseSpecifier::Name(database) = &full_name.database {
+            name_parts.push(database.as_str().into());
+        }
+        name_parts.push(full_name.schema.as_str().into());
+        name_parts.push(full_name.item.as_str().into());
+        ObjectName(name_parts)
+    }
+}
```

Comment:
Since you end up cloning the entire contents of `FullName`, it's best practice to just make the impl on the owned `FullName` like so:

```rust
impl From<FullName> for ObjectName {
    fn from(full_name: FullName) -> ObjectName {
        let mut name_parts = Vec::new();
        if let DatabaseSpecifier::Name(database) = full_name.database {
            name_parts.push(Ident::new(database));
        }
        name_parts.push(Ident::new(full_name.schema));
        name_parts.push(Ident::new(full_name.item));
        ObjectName(name_parts)
    }
}

```

That way if someone does have a `FullName` on which they're willing to give up ownership, they don't have to clone all the strings inside. And if not, you just call clone first, and you're no worse off.